pkg/endpoint/providers/kube/client.go

```diff
@@ -113,7 +112,7 @@ func (c Client) ListEndpointsForService(svc service.MeshService) []endpoint.Endp
 // GetServicesForServiceAccount retrieves a list of services for the given service account.
 func (c Client) GetServicesForServiceAccount(svcAccount service.K8sServiceAccount) ([]service.MeshService, error) {
 	log.Info().Msgf("[%s] Getting Services for service account %s on Kubernetes", c.providerIdent, svcAccount)
-	services := mapset.NewSet()
+	var services []service.MeshService
```

Comment:
The mapset was used to avoid duplicates, this needs to be handled.

Reply:
Ok, I'll change it back and just add an extra step to sort the slice by namespace/name then.

Reply:
Why does the test rely on ordering though? Could you point me to the test code which is resulting in this?

Reply:
https://github.com/nojnhuh/osm/blob/35e671640f777b1ee8ceefcee48264473b46b330/pkg/endpoint/providers/kube/client_test.go#L444

Reply:
@nojnhuh we could change the test to something like
```
		expectedServices := []{..., ...}
		Expect(services[0]).To(BeElementOf(expectedServices))
		Expect(services[1]).To(BeElementOf(expectedServices))
```

Reply:
Has the change to handle multiple services been propagated throughout though? I know the plan was to just return the first element of the list while we were working on that, so the first service should always be the same one, right?

Reply:
The change to handle multiple services has not been handled throughout, but is in progress. However, this change doesn't relate to it. We should not be introducing new assumptions that are in the process of being removed.

Either of the following work:
1. Return sorted list
2. Update the test to not rely on ordering

Reply:
Ok, I'll change the tests.